projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -177,6 +185,18 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
         return this._value;
     }
 
+    /**
+     * Specify the default input format type. Defaults to `date`, which includes
+     * only date parts for editing. Other valid options are `time` and `dateTime`.
+     *
+     * @example
+     * ```html
+     * <input igxDateTimeEditor [defaultInputFormatType]="'dateTime'">
+     * ```
+     */
+    @Input()
+    public defaultInputFormatType: DataType = DataType.Date;
```

Comment:
Besides this eliminating the need for various picker components to recalculate their own formats as discussed offline, I'm thinking this can easily be shortened to just `defaultFormatType` - unless the `displayFormat` is explicitly overridden, this will also affect it.